server/src/main/java/org/elasticsearch/cluster/routing/IndexRoutingTable.java

```diff
@@ -457,20 +458,24 @@ public Builder removeReplica() {
                 for (ShardRouting shardRouting : indexShard) {
                     builder.addShard(shardRouting);
                 }
-                // first check if there is one that is not assigned to a node, and remove it
+
+                // define remove priority
+                List<Function<ShardRouting, Boolean>> priorityRemoveClauses = List.of(
```

Comment:
Would you declare this as a `private static final` field rather than instantiating the list each time?